pkg/virt-api/api.go

```diff
@@ -918,97 +855,70 @@ func (app *virtAPIApp) createSubresourceApiservice() error {
 	return nil
 }
 
-func (app *virtAPIApp) setupTLS(fs Filesystem) error {
+func newAggregationTLSConfig(certPool *x509.CertPool, manager certificate.Manager) *tls.Config {
+	migrationTLSConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+		ClientAuth: tls.VerifyClientCertIfGiven,
+		ClientCAs:  certPool,
+		GetCertificate: func(info *tls.ClientHelloInfo) (*tls.Certificate, error) {
+			cert := manager.Current()
+			if cert == nil {
+				return nil, fmt.Errorf("no serving certificate available for virt-api")
+			}
+			return cert, nil
+		},
+	}
+	return migrationTLSConfig
+}
 
-	app.keyFile = filepath.Join(app.certsDirectory, "/key.pem")
-	app.certFile = filepath.Join(app.certsDirectory, "/cert.pem")
-	app.signingCertFile = filepath.Join(app.certsDirectory, "/signingCert.pem")
-	app.clientCAFile = filepath.Join(app.certsDirectory, "/clientCA.crt")
+func (app *virtAPIApp) startTLS() (err error) {
 
-	// Write the certs to disk
-	err := fs.WriteFile(app.clientCAFile, app.clientCABytes, 0600)
-	if err != nil {
-		return err
-	}
+	app.SetupCertificateManager(app.virtCli, func(certStore certificate.Store, name string, dnsSANs []string, ipSANs []net.IP) *certificate.Config {
+		namespacedName := fmt.Sprintf("%s.%s", virtApiServiceName, app.namespace)
+		internalAPIServerFQDN := append([]string{
+			fmt.Sprintf("%s.svc", namespacedName),
+		}, dnsSANs...)
+		return bootstrap.LoadCertConfigForService(certStore, name, internalAPIServerFQDN, ipSANs)
+	})
 
-	if len(app.requestHeaderClientCABytes) != 0 {
-		f, err := fs.OpenFile(app.clientCAFile, os.O_APPEND|os.O_WRONLY, 0600)
+	var certs []*x509.Certificate
+	if len(app.requestHeaderClientCABytes) > 0 {
```

Comment:
@davidvossel what is the difference between the HeaderClientCA and ClientCA from the apiserver? The first one works for authentication, the second one not.

Reply:
There's some info about the CAs here. https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/ I don't find this very helpful though. 

From what I remember, this is logic I took from the aggregated api server. I'm not sure about the details.

Reply:
this reminds me that I wanted to dig deeper into this CA topic for https://github.com/kubevirt/kubevirt/pull/2183...